projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_configuration_plugin.py

```diff
@@ -29,6 +29,14 @@ def vdk_configure(self, config_builder: ConfigurationBuilder) -> None:
             default_value=None,
             description="Default Ingestion target to be used.",
         )
+        config_builder.add(
+            key="INGEST_PLUGIN_EXECUTION_SEQUENCE",
```

Comment:
Not sure this is required - we should support expressions or at least a convention to determine what happens with the rest of the hook implementations that are not explicitly stated within the execution sequence. Or what happens if duplicated somewhere in the sequence. 

My understanding is, pluggy's hook impls evaluation is sequential in favour of parallel. Is that correct?

Reply:
Well, when ingestion API is used, the data engineer uses either send_object_for_ingestion() or send_tabular_data_for_ingestion(). Both of these methods require an ingestion `method` to be set (either as an env var, or passed as an argument to the method call), so there will always be at least one ingestion plugin executed.

Reply:
>Why would a data engineer register multiple ingestion plugins?

Users can have installed: 
vdk-trino - provide ingest to trino 
vdk-impala provides ingest to impala
vdk-ingest-http provides ingest ot HTTP 
And plugin that needs to convert data to a certain format (to json ld is the concrete example we have in VMware). 


And they all could be installed at the same time. 

We give options to data engineers to specify which of those plugins they want to use with configuration.
VDK_INGEST_TARGET_DEFAULT=trino


> Does Pluggy support parallelism configuration when evaluating implementations, for example .__sample_hook_spec.decorate_operation()
For this PR we will use the router functionality. Perhaps we could have written the router logic to be based on pluggy but when I was working on vdk-core initially I did not see how. Now I have some ideas but the ship has sailed for now. But I can open a ticket to look into this again.

Reply:
There seems to have been some discussions on using pluggy in the context of plugin chaining ([enhancement request](https://github.com/pytest-dev/pluggy/issues/51)), and a ([discussion](https://github.com/pytest-dev/pluggy/issues/150)), but I am not sure if anything has been actually implemented as part of pluggy.

Reply:
So plugins granularity presently is such that for example, when a data job using Impala runs, then the entire `vdk-impala` is wired. This plugin bootstraps the data source connection configuration. Also, the plugin includes ingestion payload processing capabilities. All the capabilities need to be enabled by the data job -specific configurations explicitly, is that correct? 
For example, VDK_DB_DEFAULT_TYPE=IMPALA needs to be configured to actually connect to Impala data source for executing queries purposes. What needs to be configured to pre-process the ingestion payload? What needs to be configured to actually ingest the payload on pre-processing completion?

Maybe add a test implementation, covering the Processing data job workflow? The test provides a real-world code snippet on happy path example usage. Like connect to Impala, then select data to load in-memory and transform (for example, convert to uniform lower/upper case of all string values via a test-scope implementation). Next, that data payload is automatically decorated to a specific conventional format (for example append a system timestamp property that is also auto-populated by a custom implementation in the scope of the test). After processing, that data job is submitted for ingestion via vdk-ingest-http.

Reply:
Tests added. Thanks for the feedback